src/seurat/bin/filter/sc_cell_gene_filtering.R

```diff
@@ -0,0 +1,174 @@
+    ),
+    make_option(
+        "--min-n-features",
+        type = "integer",
+        dest = "min_n_features",
+        default = -1,
+        help = "[CELL] Minimal number of features for a cell to be kept."
+    ),
+    make_option(
+        "--max-n-features",
+        type = "integer",
+        dest = "max_n_features",
+        default = -1,
+        help = "[CELL] Maximal number of features for a cell to be kept."
+    ),
+    make_option(
+        "--max-percent-mito",
+        type = "double",
+        dest = "max_percent_mito",
+        default = -1,
+        help = "[CELL] Maximal percent mitochondrial genes a cell can have."
+    ),
+    make_option(
+        "--mito-prefix",
+        type = "character",
+        dest = "mito_prefix",
+        default = "MT-",
+        help = "[CELL] Prefix to deterimine of a feature is mitochondrial or not (case sensitive!)"
+    )
+)
+
+args <- parse_args(OptionParser(option_list = option_list))
+print(args)
+
+seuratObj <- tryCatch({
+    readRDS(file = args$input)
+}, warning = function(w) {
+}, error = function(e) {
+    stop("VSN ERROR: Cannot read the given Rds file.")
+}, finally = {})
+
+if (class(x = seuratObj) != "Seurat") {
+    stop("VSN ERROR: The object contained in the Rds file is not a Seurat object.")
+}
+
+filterCells <- function(
+    object,
+    min_n_counts,
+    max_n_counts,
+    min_n_features,
+    max_n_features,
+    max_percent_mito,
+    mito_prefix
+) {
+    if (min_n_counts < 0) {
+        min_n_counts <- -Inf
+    }
+    if (max_n_counts < 0) {
+        max_n_counts <- Inf
+    }
+    if (min_n_features < 0) {
+        min_n_features <- -Inf
+    }
+    if (max_n_features < 0) {
+        max_n_features <- Inf
+    }
+    if (max_percent_mito < 0) {
+        max_percent_mito <- Inf
+    } else if (max_percent_mito < 1) {
+        max_percent_mito <- max_percent_mito * 100
+    }
+
+    mito.pattern <- paste0("^", mito_prefix)
+
+    mito.genes <- grep(mito.pattern, rownames(object))
+    if (length(mito.genes) == 0 && max_percent_mito != Inf) {
+        stop(paste0("VSN ERROR: Could not find any mitochondrial genes with the prefix: '" , mito_prefix, "'"))
+    }
+
+    object[["pct.mito"]] <- Seurat::PercentageFeatureSet(object, pattern = mito.pattern)
+    object <- subset(
+        object,
+        subset = nFeature_RNA > min_n_features & nFeature_RNA < max_n_features & nCount_RNA > min_n_counts & nCount_RNA < max_n_counts & pct.mito < max_percent_mito
+    )
+
+    return(object)
+}
+
+# FIXME: this is a very dirty approach for feature filtering
+# AFAIK it is not possible to do this in a existing object, since the features might be referenced internally (not enough to remove some rows from the matrices)
+# The current approach assumes there is no information in the object of clustering/SNN/dimred... other than some QC metadata.
+# If there woulde be other data, this will simply be lost since we only copy the metadata over to the new filtered object
```

Comment:
Agreed that this isn't the best approach, but it even seems like the Seurat team have decided not to allow filtering after creation of the object. As it's currently used, we shouldn't run into this problem so I'm not too worried at the moment.

Reply:
Agreed. I will just leave this comment in here so we don't forget about this